core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java

```diff
@@ -63,15 +51,12 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
 
         String value = null;
         float boost = 1.0f;
-        Fuzziness fuzziness = DEFAULT_FUZZINESS;
-        int prefixLength = FuzzyQuery.defaultPrefixLength;
-        int maxExpansions = FuzzyQuery.defaultMaxExpansions;
-        boolean transpositions = false;
+        Fuzziness fuzziness = FuzzyQueryBuilder.DEFAULT_FUZZINESS;
+        int prefixLength = FuzzyQueryBuilder.DEFAULT_PREFIX_LENGTH;
+        int maxExpansions = FuzzyQueryBuilder.DEFAULT_MAX_EXPANSIONS;
+        boolean transpositions = FuzzyQueryBuilder.DEFAULT_TRANSPOSITIONS;
+        String rewrite = null;
         String queryName = null;
-        MultiTermQuery.RewriteMethod rewriteMethod = null;
-        if (parseContext.isFilter()) {
-            rewriteMethod = MultiTermQuery.CONSTANT_SCORE_REWRITE;
-        }
```

Comment:
From the discussion before I think that removing the special case for FuzzyQuery inside filter context is okay here since CONSTANT_SCORE_REWRITE ist the default MultiTermQuery rewrite method anyway. 
@jpountz could you have a look at this change since I think it's related to the filter/query merging.